misc/toml-parser/src/main/java/io/ballerina/toml/semantic/ast/TomlStringValueNode.java

```diff
@@ -20,20 +20,45 @@
 
 import io.ballerina.toml.semantic.TomlType;
 import io.ballerina.toml.semantic.diagnostics.TomlNodeLocation;
-import io.ballerina.toml.syntax.tree.DocumentNode;
+import io.ballerina.toml.syntax.tree.LiteralStringLiteralNode;
+import io.ballerina.toml.syntax.tree.StringLiteralNode;
+import io.ballerina.toml.syntax.tree.SyntaxKind;
 
 /**
  * Represents A String Value in Toml AST.
  *
  * @since 2.0.0
  */
 public class TomlStringValueNode extends TomlBasicValueNode<String> {
-    public TomlStringValueNode(DocumentNode documentNode, String value, TomlNodeLocation location) {
-        super(documentNode, value, TomlType.STRING, location);
+    public TomlStringValueNode(StringLiteralNode stringLiteralNode, String value, TomlNodeLocation location) {
+        super(stringLiteralNode, value, TomlType.STRING, location);
+    }
+
+    public TomlStringValueNode(LiteralStringLiteralNode literalStringNode, String value, TomlNodeLocation location) {
+        super(literalStringNode, value, TomlType.STRING, location);
     }
 
     @Override
     public void accept(TomlNodeVisitor visitor) {
         visitor.visit(this);
     }
+
+    @Override
+    public boolean isMissingNode() {
+        if (this.externalTreeNode().kind() == SyntaxKind.STRING_LITERAL) {
```

Comment:
`this.externalTreeNode()` can be extracted to separate varaible as it is repeated.